lib/vdsm/API.py

```diff
@@ -1820,8 +1820,8 @@ def attach_volume(self, vol_id, connection_info, sd_id=None):
 
     @api.logged(on="api.storage")
     @api.method
-    def detach_volume(self, vol_id):
-        return managedvolume.detach_volume(vol_id)
+    def detach_volume(self, vol_id, sd_id=None):
+        return managedvolume.detach_volume(vol_id, sd_id=sd_id)
```

Comment:
Same comment for attach volume - use blank uuid as default, so this will be:

    def detach_volume(self, vol_id, sd_id):
        return managedvolume.detach_volume(sd_id, vol_id)